backend/de.metas.business/src/main/java/de/metas/payment/paymentterm/impl/PaymentTermRepository.java

```diff
@@ -267,16 +275,45 @@ private ImmutableListMultimap<PaymentTermId, PaymentTermBreak> retrievePaymentTe
 				.collect(GuavaCollectors.toImmutableListMultimap(termBreak -> termBreak.getId().getPaymentTermId()));
 	}
 
+	@NonNull
+	private ImmutableListMultimap<PaymentTermId, PaySchedule> retrievePayschedulesForMultipleTerms(
+			@NonNull final Collection<PaymentTermId> paymentTermIds)
+	{
+		if (paymentTermIds.isEmpty())
+		{
+			return ImmutableListMultimap.of();
+		}
+
+		return queryBL
+				.createQueryBuilder(I_C_PaySchedule.class)
+				.addOnlyActiveRecordsFilter()
+				.addInArrayFilter(I_C_PaySchedule.COLUMNNAME_C_PaymentTerm_ID, paymentTermIds)
+				.create()
+				.stream(I_C_PaySchedule.class)
+				.map(PaymentTermRepository::toPaySchedule)
+				.collect(GuavaCollectors.toImmutableListMultimap(paySched -> paySched.getId().getPaymentTermId()));
+	}
+
 	@NonNull
 	@Override
 	public ImmutableListMultimap<PaymentTermId, PaymentTermBreak> retrievePaymentTermBreaks(@NonNull final PaymentTermId paymentTermId)
 	{
 		return retrievePaymentTermBreaksForMultipleTerms(ImmutableList.of(paymentTermId));
 	}
 
+	@Override
+	public @NonNull ImmutableListMultimap<PaymentTermId, PaySchedule> retrievePaySchedules(@NonNull final PaymentTermId paymentTermId)
+	{
+		return retrievePayschedulesForMultipleTerms(ImmutableList.of(paymentTermId));
+	}
+
 	@NonNull
-	static private PaymentTerm fromRecord(@NonNull final I_C_PaymentTerm record, @NonNull final ImmutableList<PaymentTermBreak> breaks)
+	private PaymentTerm fromRecord(@NonNull final I_C_PaymentTerm record,
+								   @NonNull final ImmutableList<PaymentTermId> paymentTermIds
```

Comment:
what exactly this method will do and why it needs so many paymentTermIds just to give me one